bin/lean-gdb.py

```diff
@@ -1,3 +1,11 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+# Released under Apache 2.0 license as described in the file LICENSE.
+#
+# Author: Sebastian Ullrich
```

Comment:
`gdb bin/lean` should load the file automatically because of the file name (or print a message that the script must be whitelisted). I haven't tested it in CLion yet.

> How difficult would it be to add pretty-printers for declaration and expr?

For `expr` and a few other types, we already have top-level `print` functions. Reimplementing those in Python sounds wasteful, so while it's not the nice way, we could just call into C++ from inside the GDB pretty printer for that. For `expr`, you'd usually want to use the Lean pretty printer, which also needs an `environment`. We actually had helper functions for that and I was confused why I couldn't find them until I noticed that you had [removed](https://github.com/leanprover/lean/commit/b05b514cc2425a08a77c4c41ff4cc0bb1aabaabf#diff-be1b53035565a98b2f6d4fa12841cd36L1764) them :joy: .